apps/web/src/components/GalleryEditor/BetaAnnouncementModal.tsx

```diff
@@ -0,0 +1,108 @@
+import Image from 'next/image';
+import cover from 'public/beta-tester-announcement.png';
+import { useCallback } from 'react';
+import { graphql } from 'react-relay';
+import styled from 'styled-components';
+
+import { useModalActions } from '~/contexts/modal/ModalContext';
+import { BetaAnnouncementModalMutation } from '~/generated/BetaAnnouncementModalMutation.graphql';
+import { useReportError } from '~/shared/contexts/ErrorReportingContext';
+import { usePromisifiedMutation } from '~/shared/relay/usePromisifiedMutation';
+import colors from '~/shared/theme/colors';
+
+import { Button } from '../core/Button/Button';
+import { HStack, VStack } from '../core/Spacer/Stack';
+import { BaseXL, TitleL } from '../core/Text/Text';
+
+// TODO: Replace with actual typeform URL
+const TYPEFORM_URL = 'https://frameio.typeform.com/to/rgZ2Z4';
+
+export function BetaAnnouncementModal() {
+  const [optInForRoles] = usePromisifiedMutation<BetaAnnouncementModalMutation>(graphql`
+    mutation BetaAnnouncementModalMutation($input: [Role!]!) @raw_response_type {
+      optInForRoles(roles: $input) {
+        __typename
+        ... on OptInForRolesPayload {
+          user {
+            roles
+          }
+        }
+        ... on ErrInvalidInput {
+          __typename
+          message
+        }
+      }
+    }
+  `);
+
+  const { hideModal } = useModalActions();
+  const reportError = useReportError();
+
+  const handleContinue = useCallback(async () => {
+    hideModal();
+
```

Comment:
and also would you be able to add a track('Clicked Continue in Beta Announcement Modal') event

Reply:
Updated!